cli/ops/net.rs

```diff
@@ -137,6 +140,84 @@ fn op_accept(
   Ok(JsonOp::Async(op.boxed_local()))
 }
 
+#[derive(Deserialize)]
+struct ReceiveArgs {
+  rid: i32,
+}
+
+fn op_receive(
+  state: &State,
+  args: Value,
+  _zero_copy: Option<ZeroCopyBuf>,
+) -> Result<JsonOp, ErrBox> {
+  let args: ReceiveArgs = serde_json::from_value(args)?;
+  let rid = args.rid as u32;
+  let state_ = state.clone();
+
+  let op = async move {
+    let mut state = state_.borrow_mut();
+
+    let resource = state
+      .resource_table
+      .get_mut::<UdpSocketResource>(rid)
+      .ok_or_else(bad_resource)?;
+
+    let socket = &mut resource.socket;
+    let mut buf = vec![0; 8192];
+
+    let (_size, remote_addr) = socket.recv_from(&mut buf).await?;
```

Comment:
This is no good - `await` will hold a reference to resource until it resolves and consequently it will hold a mutable borrow to state. 

This is easy to fix though; take a look at `Accept` in `cli/ops/net.rs`. 

Something like:
```rs
pub struct Receive<'a> {
  buf: Buf,
  rid: ResourceId,
  state: &'a State,
}

impl Future for Receive<'_> {
  type Output = Result<..., ErrBox>;

  fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
    let inner = self.get_mut();
    let resource = inner.state
      .resource_table
      .get_mut::<UdpSocketResource>(rid)
      .ok_or_else(bad_resource)?;

    let socket = &mut resource.socket;

    let (_size, remote_addr) = try_ready!(socket.poll_recv_from(&mut inner.buf));
  }
}

```

Reply:
Thanks, but according to Tokio https://docs.rs/tokio/0.2.11/tokio/net/struct.UdpSocket.html#method.recv_from
There is no method poll_recv_from

Reply:
It's there, just hidden :)
https://docs.rs/tokio/0.2.11/src/tokio/net/udp/socket.rs.html#221

Reply:
Thanks, you specified a Buf type for buf, how to create a Buf?

Reply:
Found: `let mut buf = vec![0u8; 1024].into_boxed_slice();` then using `&'a mut Buf` in the struct

Reply:
@hazae41 `Buf` is just an aliast to boxed slice:
```rust
pub type Buf = Box<[u8]>;
```
So any byte vector/slice will do, just call `into_boxed_slice()` on it.